contracts/zero-ex/contracts/src/features/TransformERC20.sol

```diff
@@ -242,55 +223,65 @@ contract TransformERC20 is
 
         // Remember the initial output token balance of the taker.
         state.takerOutputTokenBalanceBefore =
-            LibERC20Transformer.getTokenBalanceOf(outputToken, taker);
+            LibERC20Transformer.getTokenBalanceOf(args.outputToken, args.taker);
 
         // Pull input tokens from the taker to the wallet and transfer attached ETH.
         _transferInputTokensAndAttachedEth(
-            inputToken,
-            taker,
+            args.inputToken,
+            args.taker,
             address(state.wallet),
-            inputTokenAmount
+            args.inputTokenAmount
         );
 
-        // Perform transformations.
-        for (uint256 i = 0; i < transformations.length; ++i) {
-            _executeTransformation(
-                state.wallet,
-                transformations[i],
-                state.transformerDeployer,
-                taker,
-                callDataHash
+        {
+            // Validate that the calldata was signed by the quote signer.
+            // `validCallDataHash` will be 0x0 if not.
+            bytes32 validCallDataHash = _getValidCallDataHash(
+                args.callDataHash,
+                args.callDataSignature
             );
+            // Perform transformations.
+            for (uint256 i = 0; i < args.transformations.length; ++i) {
+                _executeTransformation(
+                    state.wallet,
+                    args.transformations[i],
+                    state.transformerDeployer,
+                    args.taker,
+                    // Transformers will receive a null calldata hash if
```

Comment:
Curious about the benefit of passing `hash|null` opposed to a `bool`, how will the hash used by the transformer?

Reply:
My reasoning is that it comes for free and is functionally the same right now, but conceivably it could be used to track per-quote state or help augment data pipeline events.